packages/astro/test/units/vite-plugin-astro/compile.test.js

```diff
@@ -1,17 +1,17 @@
 import * as assert from 'node:assert/strict';
-import { describe, it } from 'node:test';
+import { describe, it, before } from 'node:test';
 import { pathToFileURL } from 'node:url';
 import { init, parse } from 'es-module-lexer';
 import { resolveConfig } from 'vite';
 import { compileAstro } from '../../../dist/vite-plugin-astro/compile.js';
 
-const viteConfig = await resolveConfig({ configFile: false }, 'serve');
-
+let inlineConfig;
 /**
  * @param {string} source
  * @param {string} id
  */
 async function compile(source, id) {
+	const viteConfig = await resolveConfig({ configFile: false, ...inlineConfig }, 'serve');
```

Comment:
Can we pass the `inlineConfig` as a parameter of `compile` instead? That way we avoid passing the global around.

Reply:
@bluwy Thank you for the review!

I have made the corrections in the following commit. Indeed, it seems better not to have inlineConfig as a global variable. Does this change reflect your intentions?
https://github.com/withastro/astro/pull/12676/commits/6a966c09ad590abf605fed846457b4ae30440b2b

Reply:
Perfect, thanks!